Structure_Engine/Query/Geometry.cs

```diff
@@ -108,7 +119,10 @@ private static IGeometry Geometry(this object section)
         [Deprecated("3.1", "Replaced with method for base interface IGeometricalSection.", typeof(Query), "Geometry(this IGeometricalSection section).")]
         public static IGeometry Geometry(this SteelSection section)
         {
-            return new CompositeGeometry { Elements = section.SectionProfile.Edges.ToList<IGeometry>() };
+            if (section?.SectionProfile?.Edges == null)
+                return new CompositeGeometry();
```

Comment:
Same as above, this could possibly return `null`

Reply:
Same answer too 😄